### PR TITLE
Add metric for dynamic config change

### LINKED ISSUE #7990: Add prometheus metric to display dynamic config
The dynamic config is introduced in https://github.com/near/nearcore/pull/7872.
We need a Prometheus metric for the config, such that users can easier to check and confirm the changing in the dynamic config.

_Originally posted by @mm-near in https://github.com/near/nearcore/pull/7872#discussion_r1013855953_

### PR DESCRIPTION
**Pull request by yanganto**
Fix #7990

### CODE REVIEW

File: core/dyn-configs/src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,18 @@
 #![doc = include_str!("../README.md")]
 
+use near_o11y::metrics::{try_create_int_counter, IntCounter};
+use once_cell::sync::Lazy;
 use std::sync::atomic::{AtomicU64, Ordering};
 
+/// An indicate for dynamic config changes
```
Comment (akhi3030) [inline core/dyn-configs/src/lib.rs:7]:
```suggestion
/// An indicator for dynamic config changes
```

**Review by akhi3030 — APPROVED** [review]

File: core/dyn-configs/src/lib.rs [hunk 2]
```diff
@@ -12,4 +23,5 @@ pub fn reload(expected_shutdown: Option<u64>) {
     } else {
         EXPECTED_SHUTDOWN_AT.store(0, Ordering::Relaxed);
     }
+    DYN_CONFIG_CHANGE.inc();
```
[the remark this answers is not in the document]
Comment (yanganto) [reply core/dyn-configs/src/lib.rs:27]: The document of the `reload` function is provided, and the `counting metric` is also included in the doc.

**Comment by yanganto** [comment]
Hi @akhi3030, thanks for reviewing.

**Comment by akhi3030** [comment]
@yanganto: I believe this is ready to be merged.  I will apply the merge label.  Let me know if you think otherwise.

**Pull request merged**